### ISSUE TITLE
Regression: rc0003 outputs extra characters on OSX

### ISSUE THREAD

**Issue by augustoproiete** [cake-build/cake#3077]
Cake 1.0.0-rc0003 seems to write spurious characters the first time a script writes to the Console on OSX.

I'm only able to reproduce on OSX. In Windows it works as expected (Tested Command-prompt, PowerShell, Windows Terminal).

At first glance, it appears to be a side-effect of [migrating to Spectre.Console](https://github.com/cake-build/cake/pull/3028).

### What You Are Seeing?

![image](https://user-images.githubusercontent.com/177608/106819334-07403f80-6650-11eb-80c7-fed44d3608de.png)

### What is Expected?

![image](https://user-images.githubusercontent.com/177608/106819345-0b6c5d00-6650-11eb-81b3-538ef7cd9a7f.png)

### What version of Cake are you using?

v1.0.0-rc0003

### Are you running on a 32 or 64 bit system?

64-bit

### What environment are you running on?  Windows? Linux? Mac?

OSX.

### Are you running on a CI Server?  If so, which one?

No

### How Did You Get This To Happen? (Steps to Reproduce)

- test-console.cake

```csharp
Information("Hello: {0}", "Augusto1");
Information("Hello: {0}", "Augusto2");
Information("Hello: {0}", "Augusto3");
Information("Hello: {0}", "Augusto4");
```

**Comment by patriksvensson** [comment]
@augustoproiete We're not using Spectre.Console to render anything and everything related to VT100 sequence rendering is still the same code since the last release to my knowledge.

What terminal are you running?

**Comment by augustoproiete** [comment]
![image](https://user-images.githubusercontent.com/177608/106824625-9a31a780-6659-11eb-827b-122ec1e49d84.png)

Spectre.Console was just a guess... Comparing what's changed between RC3 vs RC2, the change from Spectre.Cli to Spectre.Console is the only thing that stands out, I think:

https://github.com/cake-build/cake/compare/v1.0.0-rc0002...v1.0.0-rc0003

**Comment by augustoproiete** [comment]
Log of running with verbosity=diagnostic vs default

### Diagnostic

```
augustoproiete@mbpyt2b11 1.0.0-rc0003 % dotnet cake test-console.cake                       
Looking for modules...
No modules found to install.
Module directory does not exist.
Analyzing build script...
Analyzing /Users/augustoproiete/cake/1.0.0-rc0003/test-console.cake...
Processing build script...
Adding assembly reference to System.Private.CoreLib.dll...
Adding assembly reference to System.Linq.Expressions.dll...
Adding assembly reference to Microsoft.CSharp.dll...
Adding assembly reference to Cake.Core.dll...
Adding assembly reference to Cake.Common.dll...
Adding assembly reference to System.Runtime.dll...
Adding assembly reference to System.Collections.dll...
Adding assembly reference to System.Linq.dll...
Adding assembly reference to System.Console.dll...
Adding assembly reference to System.Text.RegularExpressions.dll...
Adding assembly reference to System.Security.Cryptography.Algorithms.dll...
Adding assembly reference to System.Diagnostics.Process.dll...
Adding assembly reference to System.IO.FileSystem.dll...
Adding assembly reference to System.Collections.Concurrent.dll...
Adding assembly reference to Microsoft.Win32.Registry.dll...
Adding assembly reference to System.Threading.dll...
Adding assembly reference to System.Security.Cryptography.Primitives.dll...
Adding assembly reference to System.Runtime.InteropServices.RuntimeInformation.dll...
Adding assembly reference to System.ComponentModel.Primitives.dll...
Adding assembly reference to System.Runtime.InteropServices.dll...
Adding assembly reference to System.Threading.Thread.dll...
Adding assembly reference to System.Xml.ReaderWriter.dll...
Adding assembly reference to System.Xml.XDocument.dll...
Adding assembly reference to System.Runtime.Serialization.Primitives.dll...
Adding assembly reference to System.Runtime.Serialization.Json.dll...
Adding assembly reference to System.Net.Http.dll...
Adding assembly reference to System.IO.Compression.dll...
Adding assembly reference to System.ComponentModel.TypeConverter.dll...
Adding assembly reference to System.Text.Encoding.Extensions.dll...
Adding assembly reference to System.Runtime.Serialization.Xml.dll...
Adding assembly reference to System.Net.Primitives.dll...
Adding assembly reference to System.IO.Compression.ZipFile.dll...
Importing namespace Cake.Common...
Importing namespace Cake.Common.Build...
Importing namespace Cake.Common.Build.AppVeyor...
Importing namespace Cake.Common.Build.AppVeyor.Data...
Importing namespace Cake.Common.Build.AzurePipelines...
Importing namespace Cake.Common.Build.AzurePipelines.Data...
Importing namespace Cake.Common.Build.Bamboo...
Importing namespace Cake.Common.Build.Bamboo.Data...
Importing namespace Cake.Common.Build.BitbucketPipelines...
Importing namespace Cake.Common.Build.BitbucketPipelines.Data...
Importing namespace Cake.Common.Build.Bitrise...
Importing namespace Cake.Common.Build.Bitrise.Data...
Importing namespace Cake.Common.Build.ContinuaCI...
Importing namespace Cake.Common.Build.ContinuaCI.Data...
Importing namespace Cake.Common.Build.GitHubActions...
Importing namespace Cake.Common.Build.GitHubActions.Data...
Importing namespace Cake.Common.Build.GitLabCI...
Importing namespace Cake.Common.Build.GitLabCI.Data...
Importing namespace Cake.Common.Build.GoCD...
Importing namespace Cake.Common.Build.GoCD.Data...
Importing namespace Cake.Common.Build.Jenkins...
Importing namespace Cake.Common.Build.Jenkins.Data...
Importing namespace Cake.Common.Build.MyGet...
Importing namespace Cake.Common.Build.TeamCity...
Importing namespace Cake.Common.Build.TravisCI...
Importing namespace Cake.Common.Build.TravisCI.Data...
Importing namespace Cake.Common.Diagnostics...
Importing namespace Cake.Common.IO...
Importing namespace Cake.Common.IO.Paths...
Importing namespace Cake.Common.Net...
Importing namespace Cake.Common.Security...
Importing namespace Cake.Common.Solution...
Importing namespace Cake.Common.Solution.Project...
Importing namespace Cake.Common.Solution.Project.Properties...
Importing namespace Cake.Common.Solution.Project.XmlDoc...
Importing namespace Cake.Common.Text...
Importing namespace Cake.Common.Tools.Cake...
Importing namespace Cake.Common.Tools.Chocolatey...
Importing namespace Cake.Common.Tools.Chocolatey.ApiKey...
Importing namespace Cake.Common.Tools.Chocolatey.Config...
Importing namespace Cake.Common.Tools.Chocolatey.Download...
Importing namespace Cake.Common.Tools.Chocolatey.Features...
Importing namespace Cake.Common.Tools.Chocolatey.Install...
Importing namespace Cake.Common.Tools.Chocolatey.New...
Importing namespace Cake.Common.Tools.Chocolatey.Pack...
Importing namespace Cake.Common.Tools.Chocolatey.Pin...
Importing namespace Cake.Common.Tools.Chocolatey.Push...
Importing namespace Cake.Common.Tools.Chocolatey.Sources...
Importing namespace Cake.Common.Tools.Chocolatey.Uninstall...
Importing namespace Cake.Common.Tools.Chocolatey.Upgrade...
Importing namespace Cake.Common.Tools.DotCover...
Importing namespace Cake.Common.Tools.DotCover.Analyse...
Importing namespace Cake.Common.Tools.DotCover.Cover...
Importing namespace Cake.Common.Tools.DotCover.Merge...
Importing namespace Cake.Common.Tools.DotCover.Report...
Importing namespace Cake.Common.Tools.DotNetCore...
Importing namespace Cake.Common.Tools.DotNetCore.Build...
Importing namespace Cake.Common.Tools.DotNetCore.BuildServer...
Importing namespace Cake.Common.Tools.DotNetCore.Clean...
Importing namespace Cake.Common.Tools.DotNetCore.Execute...
Importing namespace Cake.Common.Tools.DotNetCore.MSBuild...
Importing namespace Cake.Common.Tools.DotNetCore.NuGet.Delete...
Importing namespace Cake.Common.Tools.DotNetCore.NuGet.Push...
Importing namespace Cake.Common.Tools.DotNetCore.NuGet.Source...
Importing namespace Cake.Common.Tools.DotNetCore.Pack...
Importing namespace Cake.Common.Tools.DotNetCore.Publish...
Importing namespace Cake.Common.Tools.DotNetCore.Restore...
Importing namespace Cake.Common.Tools.DotNetCore.Run...
Importing namespace Cake.Common.Tools.DotNetCore.Test...
Importing namespace Cake.Common.Tools.DotNetCore.Tool...
Importing namespace Cake.Common.Tools.DotNetCore.VSTest...
Importing namespace Cake.Common.Tools.DupFinder...
Importing namespace Cake.Common.Tools.Fixie...
Importing namespace Cake.Common.Tools.GitLink...
Importing namespace Cake.Common.Tools.GitReleaseManager...
Importing namespace Cake.Common.Tools.GitReleaseManager.AddAssets...
Importing namespace Cake.Common.Tools.GitReleaseManager.Close...
Importing namespace Cake.Common.Tools.GitReleaseManager.Create...
Importing namespace Cake.Common.Tools.GitReleaseManager.Discard...
Importing namespace Cake.Common.Tools.GitReleaseManager.Export...
Importing namespace Cake.Common.Tools.GitReleaseManager.Label...
Importing namespace Cake.Common.Tools.GitReleaseManager.Open...
Importing namespace Cake.Common.Tools.GitReleaseManager.Publish...
Importing namespace Cake.Common.Tools.GitReleaseNotes...
Importing namespace Cake.Common.Tools.GitVersion...
Importing namespace Cake.Common.Tools.ILMerge...
Importing namespace Cake.Common.Tools.ILRepack...
Importing namespace Cake.Common.Tools.InnoSetup...
Importing namespace Cake.Common.Tools.InspectCode...
Importing namespace Cake.Common.Tools.MSBuild...
Importing namespace Cake.Common.Tools.MSpec...
Importing namespace Cake.Common.Tools.MSTest...
Importing namespace Cake.Common.Tools.NSIS...
Importing namespace Cake.Common.Tools.NuGet...
Importing namespace Cake.Common.Tools.NuGet.Add...
Importing namespace Cake.Common.Tools.NuGet.Delete...
Importing namespace Cake.Common.Tools.NuGet.Init...
Importing namespace Cake.Common.Tools.NuGet.Install...
Importing namespace Cake.Common.Tools.NuGet.List...
Importing namespace Cake.Common.Tools.NuGet.Pack...
Importing namespace Cake.Common.Tools.NuGet.Push...
Importing namespace Cake.Common.Tools.NuGet.Restore...
Importing namespace Cake.Common.Tools.NuGet.SetApiKey...
Importing namespace Cake.Common.Tools.NuGet.SetProxy...
Importing namespace Cake.Common.Tools.NuGet.Sources...
Importing namespace Cake.Common.Tools.NuGet.Update...
Importing namespace Cake.Common.Tools.NUnit...
Importing namespace Cake.Common.Tools.OctopusDeploy...
Importing namespace Cake.Common.Tools.OpenCover...
Importing namespace Cake.Common.Tools.ReportGenerator...
Importing namespace Cake.Common.Tools.ReportUnit...
Importing namespace Cake.Common.Tools.Roundhouse...
Importing namespace Cake.Common.Tools.SignTool...
Importing namespace Cake.Common.Tools.SpecFlow...
Importing namespace Cake.Common.Tools.SpecFlow.StepDefinitionReport...
Importing namespace Cake.Common.Tools.SpecFlow.TestExecutionReport...
Importing namespace Cake.Common.Tools.TextTransform...
Importing namespace Cake.Common.Tools.VSTest...
Importing namespace Cake.Common.Tools.VSWhere...
Importing namespace Cake.Common.Tools.VSWhere.All...
Importing namespace Cake.Common.Tools.VSWhere.Latest...
Importing namespace Cake.Common.Tools.VSWhere.Legacy...
Importing namespace Cake.Common.Tools.VSWhere.Product...
Importing namespace Cake.Common.Tools.WiX...
Importing namespace Cake.Common.Tools.WiX.Heat...
Importing namespace Cake.Common.Tools.XBuild...
Importing namespace Cake.Common.Tools.XUnit...
Importing namespace Cake.Common.Xml...
Importing namespace Cake.Core...
Importing namespace Cake.Core.Diagnostics...
Importing namespace Cake.Core.IO...
Importing namespace Cake.Core.Scripting...
Importing namespace System...
Importing namespace System.Collections.Generic...
Importing namespace System.IO...
Importing namespace System.Linq...
Importing namespace System.Text...
Importing namespace System.Threading.Tasks...
Compiling build script...
Hello: Augusto1
Hello: Augusto2
Hello: Augusto3
Hello: Augusto4
```

### Default

```
augustoproiete@mbpyt2b11 1.0.0-rc0003 % dotnet cake test-console.cake                       
Hello: 0mAugusto1
Hello: Augusto2
Hello: Augusto3
Hello: Augusto4
```

**Comment by devlead** [comment]
Can comfirm in iTerm 1.0.0-rc0003
<img width="179" alt="image" src="https://user-images.githubusercontent.com/1647294/106825005-4c7e6480-6684-11eb-9703-dfc684f774b6.png">
vs. 1.0.0-rc0002
<img width="124" alt="image" src="https://user-images.githubusercontent.com/1647294/106825137-80f22080-6684-11eb-85d2-795fb962da3d.png">

**Comment by devlead** [comment]
Interestingly the 0m it's a one-time thing, i.e. launch with verbose verbosity and you won't see it
<img width="266" alt="image" src="https://user-images.githubusercontent.com/1647294/106825411-1db4be00-6685-11eb-97c5-02806d17e6ce.png">

**Comment by patriksvensson** [comment]
@devlead Yeah, it's a VT100 escape seqence controlling resetting of text that gets emitted out of context for some reason: https://github.com/cake-build/cake/blob/release/1.0.0/src/Cake.Core/Diagnostics/Console/AnsiConsoleRenderer.cs#L13

**Comment by devlead** [comment]
For me behavior starts with commit d5873f0e10f76f22250e41e1a59f7919f284028c

Which is the   Spectre.Cli -> Spectre.Console migration

<img width="542" alt="image" src="https://user-images.githubusercontent.com/1647294/106827353-011a8500-6689-11eb-9989-dc4f840020c8.png">
<img width="555" alt="image" src="https://user-images.githubusercontent.com/1647294/106827792-5c4c7780-6689-11eb-811f-85c303992946.png">

**Comment by devlead** [comment]
Weirdly get the error with Spectre.Console 0.37.0 but not 0.36.0

**Comment by devlead** [comment]
Fixed by #3093
